Scraper2/Scrapper2.cpp

```diff
@@ -0,0 +1,184 @@
+                if (isGamePaused) 
+                {
+                    menu.draw(window);
+                }
+            }
+            break;
+        case sf::Event::MouseButtonPressed:
+            if (isGamePaused)
+            {
+                sf::Vector2i mousePosition = sf::Mouse::getPosition(window);
+                if (menu.getPlayButton().getGlobalBounds().contains(mousePosition.x, mousePosition.y))
+                {
+                    isGamePaused = false;
+                }
+                else if (menu.getExitButton().getGlobalBounds().contains(mousePosition.x, mousePosition.y))
+                {
+                    window.close();
+                }
+            }
+            break;
+        default:
+            break;
+        }
+    }
+}
+
+
+int main()
+{
+    srand(static_cast<unsigned int>(time(0)));
+
+    sf::RenderWindow& window = WindowHelper::Instance().GetRenderWindow();
+    GameMenu menu(window.getSize().x, window.getSize().y);
+
+    sf::Texture backgroundTexture;
+    if (!backgroundTexture.loadFromFile("back.png"))
+    {
+        std::cerr << "Error loading background texture" << std::endl;
+    }
+    sf::Sprite backgroundSprite;
+    backgroundSprite.setTexture(backgroundTexture);
+
+    Player player("wizard.png");
+    Enemy boss("boss.png", window.getSize().x - 350, window.getSize().y / 2.8);
+    sf::Clock clock;
+
+    bool isGameRunning = false;
+    bool isGamePaused = true;
+    bool playerWon = false;
+    bool gameOver = false;
+    bool resetGame = false;
+    sf::Clock gameOverClock;
+
+    window.setFramerateLimit(60);
+
+    sf::Font font;
+    if (!font.loadFromFile("ComicSansMS.ttf"))
+    {
+        std::cerr << "Error loading font" << std::endl;
+    }
+
+    sf::Text endText;
+    endText.setFont(font);
+    endText.setCharacterSize(50);
+    endText.setFillColor(sf::Color::White);
+    endText.setPosition(window.getSize().x / 4, window.getSize().y / 3);
+
+    while (window.isOpen())
+    {
+        float deltaTime = clock.restart().asSeconds();
+
+        if (gameOver)
+        {
+            if (!resetGame)
+            {
+                gameOverClock.restart();
+                resetGame = true;
+            }
+
+            if (gameOverClock.getElapsedTime().asSeconds() >= 2)
+            {
+                window.clear();
+                menu.draw(window);
+                window.display();
+
+                playerWon = false;
+                gameOver = false;
+
+                player.reset();
+                boss.reset();
+
+                isGamePaused = true;
+
+                handleSystemEvents(window, isGameRunning, isGamePaused, menu);
+                continue;
+            }
+            else
+            {
+                window.clear();
+                window.draw(backgroundSprite);
+                window.draw(endText);
+                window.display();
+
+                handleSystemEvents(window, isGameRunning, isGamePaused, menu);
```

Comment:
These function should be called regardless of the condition we are in, it's a reaction to input/system events after all, you can't just ignore that from times to times